modin/test/exchange/dataframe_protocol/omnisci/test_protocol.py

```diff
@@ -0,0 +1,259 @@
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Dataframe exchange protocol tests that are specific for OmniSci implementation."""
+
+import pytest
+import pyarrow as pa
+import pandas
+import numpy as np
+
+import modin.pandas as pd
+from modin.core.dataframe.pandas.exchange.dataframe_protocol.from_dataframe import (
+    primitive_column_to_ndarray,
+    buffer_to_ndarray,
+    set_nulls,
+)
+from modin.pandas.utils import from_arrow, from_dataframe
+from modin.pandas.test.utils import df_equals
+from modin.test.test_utils import warns_that_defaulting_to_pandas
+from .utils import get_data_of_all_types, split_df_into_chunks, export_frame
+
+
+@pytest.mark.parametrize("data_has_nulls", [True, False])
+@pytest.mark.parametrize("from_omnisci", [True, False])
+@pytest.mark.parametrize("n_chunks", [None, 3, 5, 12])
+def test_simple_export(data_has_nulls, from_omnisci, n_chunks):
+    if from_omnisci:
+        # OmniSci can't import 'uint64' as well as booleans
+        # issue for bool: https://github.com/modin-project/modin/issues/4299
+        exclude_dtypes = ["bool", "uint64"]
+    else:
+        exclude_dtypes = None
+
+    data = get_data_of_all_types(
+        has_nulls=data_has_nulls, exclude_dtypes=exclude_dtypes
+    )
+    md_df = pd.DataFrame(data)
+
+    exported_df = export_frame(md_df, from_omnisci, n_chunks=n_chunks)
+    df_equals(md_df, exported_df)
+
+
+@pytest.mark.parametrize("n_chunks", [2, 4, 7])
+@pytest.mark.parametrize("data_has_nulls", [True, False])
+def test_export_aligned_at_chunks(n_chunks, data_has_nulls):
+    """Test export from DataFrame exchange protocol when internal PyArrow table is equaly chunked."""
+    # Modin DataFrame constructor can't process PyArrow's category when using `from_arrow`, so exclude it
+    data = get_data_of_all_types(has_nulls=data_has_nulls, exclude_dtypes=["category"])
+    pd_df = pandas.DataFrame(data)
+    pd_chunks = split_df_into_chunks(pd_df, n_chunks)
+
+    chunked_at = pa.concat_tables([pa.Table.from_pandas(pd_df) for pd_df in pd_chunks])
+    md_df = from_arrow(chunked_at)
+    assert (
+        len(md_df._query_compiler._modin_frame._partitions[0][0].get().column(0).chunks)
+        == md_df.__dataframe__().num_chunks()
+        == n_chunks
+    )
+
+    exported_df = export_frame(md_df)
+    df_equals(md_df, exported_df)
```

Comment:
these lines is a common place where you would put a breakpoint in order to check why `df_equals` is failing. It would be more convenient to just print/explore already computed `exported_df` rather than recomputing it all the time you rerun the flow.

Reply:
Makes sense. Then I suggest naming the results of `export_frame()` calls differently instead of re-assigning a single variable all over - this way it would read better and would be less error-prone for copy-paste problems if a new case would be added.

Reply:
IMHO different variable names approach is much more affected by copy-paste problems as it would require more modifications in order to add a new case:
1. Different variable names approach:
    ```diff
        exported_df1 = export_frame(md_df)
        df_equals(exported_df1, md_df)
    +
    +  # I've added another test case by copy-pasting the previous one
    +   exported_df2 = export_frame(md_df.post_processing()) # Changed the variable name here
    +   df_equals(exported_df1, md_df) # Forgot to do the same here, as the result new test case tests nothing
    ```
2. Re-assigning the same variable approach:
    ```diff
        exported_df = export_frame(md_df)
        df_equals(exported_df, md_df)
    +
    +   # I've added another test case by copy-pasting the previous one with minimal modifications
    +   exported_df = export_frame(md_df.post_processing())
    +   df_equals(exported_df, md_df)
    ```